Add `#[HandlerFor]` Attribute

adrian-enspired opened the pull request:
Supports declaring a resolver that registers command handlers and explicitly mapping the commands each handles.

This is the opposite approach to using the `#[HandledBy]` attribute, which allows commands to specify their handler.
Usage is like:
```php
<?php

use Brammm\Tactishun\CommandBus;
use Brammm\Tactishun\HandlerFor;
use Brammm\Tactishun\HandlerForResolver;
use Brammm\Tactishun\CommandHandler;

// 1. Create your command
final readonly class SendWelcomeMail {
    public function __construct( public UserId $userId ) {}
}

// 2. Implement the handler
/** @implements CommandHandler<SendWelcomeMail> */
#[HandlerFor(SendWelcomeEmail::class)]
final class SendWelcomeMailHandler implements CommandHandler {
    public function handle(object $command): void {
        // Fetch user from $command->userId and send welcome email
    }
}

// 3. Set up the command bus
$container = new Container(); // Any PSR-11 compatible container
$resolver = new HandlerForResolver(SendWelcomeMailHandler::class);
$commandBus = new CommandBus($container, $resolver);

// 4. Dispatch commands
$commandBus->handle(new SendWelcomeMail($user->id));
```

Brammm commented:
Thanks for this! I was working on something similar that I'll probably merge with this approach. I'm working on Psalm integration as well, which is making it a little more difficult than I'd like.

adrian-enspired commented:
rebased on main.

added a `ResolverAggregate` that allows multiple resolvers to be attempted in turn.
```php
<?php

$resolverA = new HandledByCommandHandlerResolver();
$resolverB = new HandlerForResolver(SendWelcomeMailHandler::class);
$aggregateResolver = new ResolverAggregate($resolverA, $resolverB);
$commandBus = new CommandBus($container, $aggregateResolver);
```

i can separate this out into another PR if needed.
